staging/src/k8s.io/kubectl/pkg/drain/cordon.go

```diff
@@ -99,13 +100,20 @@ func (c *CordonHelper) PatchOrReplaceWithContext(clientCtx context.Context, clie
 		if serverDryRun {
 			patchOptions.DryRun = []string{metav1.DryRunAll}
 		}
-		_, err = client.Patch(clientCtx, c.node.Name, types.StrategicMergePatchType, patchBytes, patchOptions)
+		updatedNode, err := client.Patch(clientCtx, c.node.Name, types.StrategicMergePatchType, patchBytes, patchOptions)
+		if err == nil {
+			c.node.Spec.Unschedulable = updatedNode.Spec.Unschedulable
```

Comment:
I think, there is another issue in drain/cordon.go. We shouldn't update `node.Spec.Unschedulable`, if `serverDryRun` is true.